### PR TITLE
Don't add errored parsed results

### LINKED ISSUE #2777: Error on the Translators table
image
https://de.openfoodfacts.org/cgi/top_translators.pl

### PR DESCRIPTION
**Pull request by hangy**
**Description:**
Only add rows to the `dataSource` if there were no errors parsing the CSV row.

**Related issues and discussion:** Fixes #2777

### CODE REVIEW

**Comment by VaiTon** [comment]
@hangy what happens if there are errors?

**Comment by hangy** [comment]
> @hangy what happens if there are errors?

The papaparse returns the last row as containing errors, because of missing columns. The row object `results.data` only contains the `Name` property with `null` as the value. All other properties/columns are missing, which in turn causes DataTables to trip over the data. This is only exposed by `results.errors` containing an error object.

**Comment by VaiTon** [comment]
> 
> 
> > @hangy what happens if there are errors?
> 
> The papaparse returns the last row as containing errors, because of missing columns. The row object `results.data` only contains the `Name` property with `null` as the value. All other properties/columns are missing, which in turn causes DataTables to trip over the data. This is only exposed by `results.errors` containing an error object.

So, if I understand correctly, errored results aren't added to the table. Does it show an error message in the console? We could do that to inform the power user that something went wrong.

**Comment by hangy** [comment]
> So, if I understand correctly, errored results aren't added to the table. Does it show an error message in the console? We could do that to inform the power user that something went wrong.

ACK. I've added a warning log message in dc0f4ecef0a65707a795d2dab78bf2fd547a5d9b.

File: cgi/top_translators.pl [hunk 1]
```diff
@@ -63,6 +63,12 @@
 				if (results.errors.length === 0) {
 					dataSource = dataSource.concat(results.data);
 				}
+				else {
+					for (var i = 0; i < results.errors.length; ++i) {
```
Comment (VaiTon) [inline cgi/top_translators.pl]: I would use a forEach:
```js
results.errors.forEach(error => {
  console.warn("...");
}
```

Comment (hangy) [reply cgi/top_translators.pl]: There's no huge advantage over the `for` loop, and arrow functions don't work in a few browsers. I'll leave it as-is for now.

**Pull request merged**